app/models/movie.rb

```diff
@@ -1,3 +1,9 @@
 class Movie < ApplicationRecord
+  has_many :read_texts, dependent: :destroy
+  has_many :read_texted_users, through: :likes, source: :user
   validates :title, :url, presence: true
+
+  def read_text_by?(user)
+    read_texts.find_by(user_id: user.id).present?
+  end
```

Comment:
AWS教材の読破済みを実装するのに動画の方にコードを書いてしまっていますよ。`app/models/aws_text.rb` に移動しましょう。

```suggestion
  validates :title, :url, presence: true
```